### PR TITLE
Use the right way to check if an argument has fixed type

### PR DESCRIPTION
**Pull request by liqunl**
We should ask `classIsFixed()` on the argument to check if it has fixed class constraint.

Signed-off-by: Liqun Liu <liqunl@ca.ibm.com>

### CODE REVIEW

**Comment by liqunl** [comment]
@andrewcraik Could you review?

**Comment by andrewcraik** [comment]
Looks good to me - @vijaysun-omr can you take a look?

File: compiler/optimizer/LocalOpts.cpp [hunk 1]
```diff
@@ -7462,18 +7461,12 @@ int32_t TR_InvariantArgumentPreexistence::perform()
                   }
                else
                   {
-                  clazz = fe()->getClassFromSignature(sig, len, feMethod);
+                  clazz = clazz ? clazz : fe()->getClassFromSignature(sig, len, feMethod);
```
Comment (vijaysun-omr) [inline compiler/optimizer/LocalOpts.cpp]: Should you be taking the more specific of clazz or getClassFromSignature ? i.e. maybe both are non-null but one provides "better" type info than the other (?)

Comment (liqunl) [reply compiler/optimizer/LocalOpts.cpp]: Inliner will try to refine argument types and put the result in argInfo. The type from argInfo should be the same as or more specific than the type from method symbol.

Comment (andrewcraik) [reply compiler/optimizer/LocalOpts.cpp]: should we have an assert for that then maybe just to verify it is true and warn ourselves is something breaks that invariant unintentionally later on?

Comment (liqunl) [reply compiler/optimizer/LocalOpts.cpp]: Assertion is added

**Comment by vijaysun-omr** [comment]
@genie-omr build all

**Pull request merged**